pkg/client/database/postgres/postgres.go

```diff
@@ -0,0 +1,294 @@
+		return nil
+	})
+}
+
+func (c postgresClient) ParseInputSecret(ctx context.Context, postgres v1alpha1.Postgres) (string, error) {
+	if postgres.Spec.ForProvider.MasterPasswordSecretRef == nil {
+		return "", errors.New(errGetPasswordSecretFailed)
+	}
+	nn := types.NamespacedName{
+		Name:      postgres.Spec.ForProvider.MasterPasswordSecretRef.Name,
+		Namespace: postgres.Spec.ForProvider.MasterPasswordSecretRef.Namespace,
+	}
+	s := &v1.Secret{}
+	if err := c.kube.Get(ctx, nn, s); err != nil {
+		return "", errors.Wrap(err, errGetPasswordSecretFailed)
+	}
+	return string(s.Data[postgres.Spec.ForProvider.MasterPasswordSecretRef.Key]), nil
+}
+
+func MakePVCPostgres(postgres *v1alpha1.Postgres) (*v1.PersistentVolumeClaim, error) {
+	fs := v1.PersistentVolumeFilesystem
+	quan, err := resource.ParseQuantity(postgres.Spec.ForProvider.DatabaseSize)
+	if err != nil {
+		return nil, err
+	}
+	return &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      postgres.Name,
+			Namespace: postgres.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PersistentVolumeClaim",
+			APIVersion: "v1",
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes:      []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+			VolumeMode:       &fs,
+			StorageClassName: postgres.Spec.ForProvider.StorageClass,
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceStorage: quan,
+				},
+			},
+		},
+	}, nil
+}
+
+func MakePostgresDeployment(ps *v1alpha1.Postgres, pw string) *appsv1.Deployment {
+	depl := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ps.Name,
+			Namespace: ps.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Replicas: utils.Int32(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"deployment": ps.Name,
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Volumes: []v1.Volume{
+						{
+							Name: ps.Name,
+							VolumeSource: v1.VolumeSource{
+								PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+									ClaimName: ps.Name,
+								},
+							},
+						},
+					},
+					Containers: MakeDefaultPostgresPodContainers(ps, pw),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"deployment": ps.Name,
+					},
+				},
+			},
+		},
+	}
+	// required for restricted namespace
+	if strings.HasPrefix(ps.Namespace, NamespacePrefixOpenShift) {
+		userGroupId := int64(26)
+		depl.Spec.Template.Spec.SecurityContext = &v1.PodSecurityContext{
+			FSGroup:            &userGroupId,
+			SupplementalGroups: []int64{userGroupId},
+		}
+	}
+	return depl
+}
+
+func MakeDefaultPostgresPodContainers(ps *v1alpha1.Postgres, pw string) []v1.Container {
+	return []v1.Container{
+		{
+			Name:  ps.Name,
+			Image: "postgres:13.0",
```

Comment:
Does this need to be hard coded? If so, and if it's used in a few places, maybe a constant can be used? If it's only used in this single place it's probably fine.

Reply:
For now, it is only used in this place, but I'll move it to a constant.